cyclonedx/model/__init__.py

```diff
@@ -147,13 +226,43 @@ class ExternalReferenceType(Enum):
     LICENSE = 'license'
     MAILING_LIST = 'mailing-list'
     OTHER = 'other'
+    RELEASE_NOTES = 'release-notes'  # Only supported in >= 1.4
     SOCIAL = 'social'
     SCM = 'vcs'
     SUPPORT = 'support'
     VCS = 'vcs'
     WEBSITE = 'website'
 
 
+class XsUri:
+    """
+    Helper class that allows us to perform validation on data strings that are defined as xs:anyURI
+    in CycloneDX schema.
+
+    Developers can just use this via `str(XsUri('https://www.google.com'))`.
+
+    .. note::
+        See XSD definition for xsd:anyURI: http://www.datypic.com/sc/xsd/t-xsd_anyURI.html
+    """
+
+    invalid_uri_regex = re.compile("(%(?![0-9A-F]{2})|#.*#)", re.IGNORECASE + re.MULTILINE)
```

Comment:
remove the outer catching brackets `()`. the finding itself seams not to be used.
`r'%(?![0-9A-F]{2})|#.*#'` shoud be fine

Reply:
better: make the property a private static constant.

Reply:
Sorted.